src/media-segment-request.js

```diff
@@ -235,59 +248,235 @@ const handleSegmentResponse = (segment, captionParser, finishProcessingFn) => (e
   segment.stats = getRequestStats(request);
 
   if (segment.key) {
-    segment.encryptedBytes = new Uint8Array(request.response);
+    segment.encryptedBytes = new Uint8Array(newBytes);
   } else {
-    segment.bytes = new Uint8Array(request.response);
+    segment.bytes = new Uint8Array(newBytes);
+  }
+
+  return finishProcessingFn(null, segment);
+};
+
+const transmuxAndNotify = ({
+  segment,
+  bytes,
+  isPartial,
+  trackInfoFn,
+  timingInfoFn,
+  id3Fn,
+  captionsFn,
+  dataFn,
+  doneFn
+}) => {
+  // Keep references to each function so we can null them out after we're done with them.
+  // One reason for this is that in the case of full segments, we want to trust start
+  // times from the probe, rather than the transmuxer.
+  let audioStartFn = timingInfoFn.bind(null, segment, 'audio', 'start');
+  let audioEndFn = timingInfoFn.bind(null, segment, 'audio', 'end');
+  let videoStartFn = timingInfoFn.bind(null, segment, 'video', 'start');
+  let videoEndFn = timingInfoFn.bind(null, segment, 'video', 'end');
+
+  // Check to see if we are appending a full segment.
+  if (!isPartial && !segment.lastReachedChar) {
+    // In the full segment transmuxer, we don't yet have the ability to extract a "proper"
+    // start time. Meaning cached frame data may corrupt our notion of where this segment
+    // really starts. To get around this, full segment appends should probe for the info
+    // needed.
+    const probeResult = probeTsSegment(bytes, segment.baseStartTime);
+
+    if (probeResult) {
+      trackInfoFn(segment, {
```

Comment:
You could probably just pass `probeResult` here `trackInfoFn(segment, probeResult);`

Reply:
I think I was avoiding passing the whole object so we could me more explicit about which parameters we are passing, but open to the alternative if you prefer it.